concordium-std/src/types.rs

```diff
@@ -581,6 +584,93 @@ pub struct ExternReturnValue {
     pub(crate) current_position: u32,
 }
 
+/// The current public balances of an account.
+#[derive(Debug, Copy, Clone)]
+pub struct AccountBalance {
+    /// The total balance of the account. Note that part of this balance might
+    /// be staked and/or locked in scheduled transfers.
+    pub(crate) total:  Amount,
+    /// The current staked amount of the account. This amount is used for
+    /// staking.
+    pub(crate) staked: Amount,
+    /// The current amount locked in releases that resulted from transfers with
+    /// schedule. A locked amount can still be used for staking.
+    pub(crate) locked: Amount,
+}
+
+impl AccountBalance {
+    /// Construct a new account balance.
+    pub fn new(total: Amount, staked: Amount, locked: Amount) -> Self {
+        Self {
+            total,
+            staked,
+            locked,
+        }
+    }
+
+    /// The total balance of the account. Note that part of this balance might
+    /// be staked and/or locked in releases using scheduled transfers.
+    pub fn total(&self) -> Amount { self.total }
+
+    /// The current staked amount of the account. This amount is used for
+    /// staking.
+    pub fn staked(&self) -> Amount { self.staked }
+
+    /// The current amount locked in releases that resulted from transfers with
+    /// schedule. A locked amount can still be used for staking.
+    pub fn locked(&self) -> Amount { self.locked }
+
+    /// The current available balance of the account. This is the amount
+    /// an account currently have available for transfering and is not
+    /// staked or locked in releases by scheduled transfers.
+    pub fn available(&self) -> Amount { self.total - max(self.locked, self.staked) }
```

Comment:
This might underflow due to the missing check in `new`.

Reply:
Good catch